test/FunctionalTests/Infrastructure/ApplicationTestFixture.cs

```diff
@@ -78,35 +98,89 @@ public void Dispose()
                 _deploymentTask.Result.HttpClient?.Dispose();
             }
 
+            CleanupWorkingDirectory();
+
             _deployer?.Dispose();
         }
 
-        public async Task<DeploymentResult> CreateDeploymentAsync(ILoggerFactory loggerFactory)
+        public Task<DeploymentResult> CreateDeploymentAsync(ILoggerFactory loggerFactory)
         {
-            try
+            if (_deploymentTask == null)
             {
-                await _deploymentLock.WaitAsync(TimeSpan.FromSeconds(10));
-                if (_deploymentTask == null)
-                {
-                    var deploymentParameters = GetDeploymentParameters();
-                    if (PublishOnly)
-                    {
-                        _deployer = new PublishOnlyDeployer(deploymentParameters, loggerFactory);
-                    }
-                    else
-                    {
-                        _deployer = ApplicationDeployerFactory.Create(deploymentParameters, loggerFactory);
-                    }
+                _deploymentTask = CreateDeploymentAsyncCore(loggerFactory);
+            }
+
+            return _deploymentTask;
+        }
+
+        protected virtual Task<DeploymentResult> CreateDeploymentAsyncCore(ILoggerFactory loggerFactory)
```

Comment:
Should get us out of some file lock contentions since we build a copy of the project and not a shared copy. Also lets us write tests varying by parameters such as `Platform`